test/pricer_rule/pay.js

```diff
@@ -202,33 +229,33 @@ contract('PricerRule::pay', async () => {
                 tokenRules,
                 conversionRate,
                 conversionRateDecimals,
+                requiredPriceOracleDecimals,
                 pricerRule,
                 quoteCurrencyCode,
-                priceOracleInitialPrice,
                 priceOracle,
                 fromAddress,
             } = await prepare(accountProvider);
 
-            const oraclePrice = 2;
+            const oraclePriceBN = new BN(4 * (10 ** requiredPriceOracleDecimals));
             await priceOracle.setPrice(
-                oraclePrice,
+                oraclePriceBN.toNumber(),
```

Comment:
Why do you convert `toNumber()`? If `priceOracle` is a contract, web3 will convert it to BN again 🤔